src/types/colorTable.cpp

```diff
@@ -0,0 +1,1073 @@
+    std::pair{ "yellow1"sv, til::color{ 255, 255, 0 } },
+    std::pair{ "yellow2"sv, til::color{ 238, 238, 0 } },
+    std::pair{ "yellow3"sv, til::color{ 205, 205, 0 } },
+    std::pair{ "yellow4"sv, til::color{ 139, 139, 0 } },
+    std::pair{ "yellowgreen"sv, til::color{ 154, 205, 50 } }
+};
+
+// Function Description:
+// - Fill the first 16 entries of a given color table with the Campbell color
+//   scheme, in the ANSI/VT RGB order.
+// Arguments:
+// - table: a color table with at least 16 entries
+// Return Value:
+// - <none>, throws if the table has less that 16 entries
+void Utils::InitializeCampbellColorTable(const gsl::span<COLORREF> table)
+{
+    THROW_HR_IF(E_INVALIDARG, table.size() < 16);
+
+    std::copy(campbellColorTable.begin(), campbellColorTable.end(), table.begin());
+}
+
+// Function Description:
+// - Fill the first 16 entries of a given color table with the Campbell color
+//   scheme, in the Windows BGR order.
+// Arguments:
+// - table: a color table with at least 16 entries
+// Return Value:
+// - <none>, throws if the table has less that 16 entries
+void Utils::InitializeCampbellColorTableForConhost(const gsl::span<COLORREF> table)
+{
+    THROW_HR_IF(E_INVALIDARG, table.size() < 16);
+    InitializeCampbellColorTable(table);
+    SwapANSIColorOrderForConhost(table);
+}
+
+// Function Description:
+// - modifies in-place the given color table from ANSI (RGB) order to Console order (BRG).
+// Arguments:
+// - table: a color table with at least 16 entries
+// Return Value:
+// - <none>, throws if the table has less that 16 entries
+void Utils::SwapANSIColorOrderForConhost(const gsl::span<COLORREF> table)
+{
+    THROW_HR_IF(E_INVALIDARG, table.size() < 16);
+    std::swap(til::at(table, 1), til::at(table, 4));
+    std::swap(til::at(table, 3), til::at(table, 6));
+    std::swap(til::at(table, 9), til::at(table, 12));
+    std::swap(til::at(table, 11), til::at(table, 14));
+}
+
+// Function Description:
+// - Fill the first 255 entries of a given color table with the default values
+//      of a full 256-color table
+// Arguments:
+// - table: a color table with at least 256 entries
+// Return Value:
+// - <none>, throws if the table has less that 256 entries
+void Utils::Initialize256ColorTable(const gsl::span<COLORREF> table)
+{
+    THROW_HR_IF(E_INVALIDARG, table.size() < 256);
+
+    std::copy(standardXterm256ColorTable.begin(), standardXterm256ColorTable.end(), table.begin());
+}
+
+#pragma warning(push)
+#pragma warning(disable : 26447) // This is a false positive.
+
+// Function Description:
+// - Parses a color from a string based on the XOrg app color name table.
+// Arguments:
+// - str: a string representation of the color name to parse
+// Return Value:
+// - An optional color which contains value if a color was successfully parsed
+std::optional<til::color> Utils::ColorFromXOrgAppColorName(const std::wstring_view wstr) noexcept
+try
+{
+    std::wstring key(wstr);

```

Comment:
Why take a local copy of the string if you're just going to walk it character-by-character anyway?

Reply:
Yeah...it's because originally I used `std::transform` to towlower it so I needed a mutable copy of it. Now it's not really needed.